front/src/applications/stdcm/hooks/useConflicts.ts

```diff
@@ -0,0 +1,108 @@
+import { useEffect, useState } from 'react';
+
+import type { TFunction } from 'i18next';
+
+import type { ManageTrainSchedulePathProperties } from 'applications/operationalStudies/types';
+import { processConflicts } from 'applications/stdcm/utils/fomatConflicts';
+import type { Conflict } from 'common/api/osrdEditoastApi';
+
+const useConflictsMessages = (
+  t: TFunction,
+  conflictData?: Conflict[],
+  pathProperties?: ManageTrainSchedulePathProperties
+) => {
+  const [trackConflicts, setTrackConflicts] = useState<string[]>([]);
+  const [workConflicts, setWorkConflicts] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (!pathProperties || !conflictData) return;
+
+    const generateConflictMessages = () => {
+      const processedConflictsData = processConflicts(conflictData, pathProperties);
+
+      const trackConflictsData = processedConflictsData.filter(
+        (conflict) => conflict.conflictType === 'Spacing'
+      );
+
+      const workConflictsData = processedConflictsData.filter(
+        (conflict) => conflict.conflictType !== 'Spacing'
+      );
```

Comment:
The type of conflict doesn't mean it will be a track or work conflict, we should use `work_schedule_ids`. If there are ids inside, we know it will be a work conflict.

Reply:
Handled like discussed :)